src/Ccap/Codegen/Scala.purs

```diff
@@ -184,6 +189,17 @@ typeDecl outputMode (TypeDecl name tt an) =
               (((text ("Decoder.enum[M, " <> name) <<>> char ']') `paren` params) // text ".disjunction")
       pure $ trait // ((text "object" <<+>> text name) `curly` variants) // enc // dec
 
+fieldNames :: Maybe String -> Array String -> Box
+fieldNames mod names =
+  maybe body (\m -> curly (text "object" <<+>> text m) [ body ]) mod
+  where
+    body = curly (text "object" <<+>> text "FieldNames") (names <#> fieldNameConst)
+    fieldNameConst s =
+      text "val" <<+>> text (valName s) <<>> text ": String" <<+>> text "=" <<+>> text (show s)
+    valName s =
+      let { before, after } = String.splitAt 1 s
+      in String.toUpper before <> after
```

Comment:
I guess we have a trade-off there wrt using standard Scala constant naming convention vs. matching the field names vs. outputting both. What are your thoughts on that?

Reply:
I was thinking we'd follow the standard for constants, but I don't have strong opinions here and am happy to change this. I'd expect we'd get used to whatever is used there. Outputting both does sound a little messy though.